client/js/components/MapView/MapView.js

```diff
@@ -0,0 +1,83 @@
+import React, { Component } from "react";
+import { TouchableOpacity } from "react-native";
+import MapView, { PROVIDER_GOOGLE } from "react-native-maps";
+import styles from "./styles";
+
+class MapViewComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      latitude: 49.2827,
+      longitude: 123.1207,
+      error: null
+    };
+  }
+
+  componentDidMount() {
+    navigator.geolocation.getCurrentPosition(position => {
+      this.setState({
+        latitude: position.coords.latitude,
+        longitude: position.coords.longitude,
+        error: null
+      });
+    });
+  }
+
+  getCurrentLocation = () => {
+    this.watchID = navigator.geolocation.watchPosition(
+      position => {
+        this.setState({
+          latitude: position.coords.latitude,
+          longitude: position.coords.longitude,
+          error: null
+        });
+      },
+      error => console.log(error),
+      {}
+    );
+  };
+
+  componentDidUpdate(prevProps) {
+    navigator.geolocation.clearWatch(this.watchID);
+    navigator.geolocation.stopObserving();
+    if (this.props.searchedLocation !== prevProps.searchedLocation) {
+      this.setState({
+        latitude: this.props.searchedLocation.lat,
+        longitude: this.props.searchedLocation.lng
+      });
+    }
+  }
+
+  render() {
+    const { latitude, longitude } = this.state;
+    return (
+      <MapView
+        provider={PROVIDER_GOOGLE}
+        style={styles.mapView}
+        region={{
+          latitude: latitude,
+          longitude: longitude,
+          latitudeDelta: 0.0922,
+          longitudeDelta: 0.0421
+        }}
+        initialRegion={{
+          latitude: 49.2827,
```

Comment:
Remove the hard-code latitude and longitude values.

Reply:
There has to be initial region latitude and longitude values other wise the map will show up somewhere else